src/parameterizeRoute.js

```diff
@@ -12,6 +12,6 @@ module.exports = function (string, params={}) {
       throw TypeError(`"${key}" was not provided in the given params for ${string}`)
     }
 
-    return params[key] || ''
+    return params[key] !== undefined ? params[key] : ''
```

Comment:
What about a simple membership check here, like `key in params`? We'd get around type checking and we can't really prevent bad values from passing through anyway.